kubernetes/helm/pinot/templates/controller/ingress.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.controller.ingress.enabled -}}
+{{- $ingressPath := .Values.controller.ingress.path -}}
+{{- $serviceName := include "pinot.controller.fullname" . -}}
+{{- $servicePort := .Values.controller.service.port -}}
+apiVersion: extensions/v1beta1
```

Comment:
is this `extensions/v1beta1` or `networking.k8s.io/v1`?
Ref: https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource

Reply:
@xiangfu0 as commented on Slack, the project wants to 
>catch up with most the cloud vendors. So far the lowest version is 1.18.16 from Amazon EKS

And `extensions/v1beta1` is supported out of the box from 1.18 forward while `networking.k8s.io/v1` is supported from `1.19`. I guess we want to keep it as `extensions/v1beta1`?

Reply:
Sounds good. 

I also checked that this `extensions/v1beta1` is deprecated on 1.22.  https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122

How about change this ingress file from `ingress.yaml` to `ingress-v1beta1.yaml`, and make the flag to enable it to `.Values.controller.ingress.v1beta1.enabled`? So we can add another `ingress.yaml` file with `networking.k8s.io/v1` api.

Reply:
or we can just keep the file name still `ingress.yaml` and use flags `.Values.controller.ingress.v1beta1.enabled`, .Values.controller.ingress.v1.enabled` to switch between two APIs.

Reply:
@xiangfu0 I added the sections `.Values.controller.ingress.v1beta1` and `.Values.controller.ingress.v1`.

Let me know if that addresses your point.